onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -210,10 +210,18 @@ void addOrtValueMethods(pybind11::module& m) {
 #ifdef ENABLE_TRAINING
       .def("to_dlpack", [](OrtValue* ort_value) -> py::object {
         return py::reinterpret_steal<py::object>(ToDlpack(*ort_value));
-      })
+      }, "Returns a DLPack representing the tensor.")
       .def_static("from_dlpack", [](py::object data, bool is_bool_tensor = false) {
         return FromDlpack(data.ptr(), is_bool_tensor);
-      })
+      }, "Convers a tensor from a external library into an OrtValue by means of the __dlpack__ protocol.")
+      .def("__dlpack__", [](OrtValue* ort_value) -> py::object {
+        return py::reinterpret_steal<py::object>(ToDlpack(*ort_value));
+       }, "Returns a DLPack representing the tensor (part of __dlpack__ protocol).")
+      .def("__dlpack_device__", [](OrtValue* ort_value) -> py::tuple {
```

Comment:
Maybe we can get this info from OrtValue directly instead of converting it to DLPack first.

Reply:
We could do that if OrtValue was using DLPack structure to describe a tensor but it is not the case. So we need to create this intermediate structure. pytorch does the same: https://github.com/pytorch/pytorch/blob/master/torch/csrc/Module.cpp#L365, https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/DLConvertor.cpp#L217.

Reply:
I think I am talking about the __dlpack_device__, PyTorch implements it here: https://github.com/pytorch/pytorch/blob/88d86de7d8887660e3277b06c49147c4e0df5b89/torch/_tensor.py#L1156. And it gets the info from the torch tensor (the self arugment) directly, though this is in Python code. Our code uses pybind and the code is in C++, but it's same thing.

Reply:
I removed the creation of a new structure and only called the necessary code to return the device.